firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift

```diff
@@ -401,4 +407,8 @@ final class ToolbarMiddleware: FeatureFlaggable {
 
         toolbarTelemetry.readerModeButtonTapped(isPrivate: toolbarState.isPrivateMode, isEnabled: isReaderModeEnabled)
     }
+
+    private func tabManager(for uuid: WindowUUID) -> TabManager {
+        return (AppContainer.shared.resolve() as WindowManager).tabManager(for: uuid)
```

Comment:
It would be better to have the `WindowManager` injectable for easier testing. Because of this we should add it as a parameter for the init.